Library/Services/Implementation/NfieldSurveyResourcesService.cs

```diff
@@ -0,0 +1,66 @@
+﻿//    This file is part of Nfield.SDK.
+//
+//    Nfield.SDK is free software: you can redistribute it and/or modify
+//    it under the terms of the GNU Lesser General Public License as published by
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using Newtonsoft.Json;
+using Nfield.Extensions;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Nfield.Services.Implementation
+{
+    /// <summary>
+    /// Implementation of <see cref="INfieldSurveyResourcesService"/>
+    /// </summary>
+    internal class NfieldSurveyResourcesService : INfieldSurveyResourcesService, INfieldConnectionClientObject
+    {
+        #region Implementation of INfieldSurveyResourcesService
+
+        private Uri SurveyResourcesApi => new Uri(ConnectionClient.NfieldServerUri, "SurveyResources/");
+
+        /// <summary>
+        /// See <see cref="INfieldSurveyResourcesService.QueryAsync"/>
+        /// </summary>
+        public Task<IQueryable<SurveyResources>> QueryAsync()
+        {
+            return Client.GetAsync(SurveyResourcesApi)
+                         .ContinueWith(
+                             responseMessageTask => responseMessageTask.Result.Content.ReadAsStringAsync().Result)
+                         .ContinueWith(
+                             stringTask =>
+                             JsonConvert.DeserializeObject<List<SurveyResources>>(stringTask.Result).AsQueryable())
+                         .FlattenExceptions();
```

Comment:
You say "supports odata filtering" but that is not what is happening here; all resources are always fetched and the filter is applied client-side.

Reply:
Yes, I added the odata filtering just to the controller in the API. It wasn't a requirement but It was easy and fast to add it so I did it.  Filtering could be used here as well, i'm going to check what I need to add it.

Reply:
Don't worry about it too much, it doesn't seem like a huge deal. The surveys endpoint works the same way.